File: application/Domain/Entities/Pet.cs
namespace application.Domain.Entities;

public class Pet
{
    public int Id { get; set; }

    public required string Name { get; set; }

    public required string Species { get; set; }

    public required string Breed { get; set; }

    public ICollection<Color> Colors { get; set; } = new HashSet<Color>();

    public ICollection<WeightRegistry> WeightRegistries { get; set; } = new HashSet<WeightRegistry>();

    public ICollection<EventAnnotation> EventAnnotations { get; set; } = new HashSet<EventAnnotation>();

    public DateTime Birth { get; set; }

    public DateTime? Death { get; set; }

    public DateTime Adoption { get; set; }

    public required string ImgContent { get; set; }
}

File: angularapp/src/app/pets/models/pet.ts
import { EventAnnotation } from "src/app/event-list/models/eventAnnotation";
import { Weight } from "src/app/weight-list/models/weight";

export interface Pet {
  id: number;
  name: string;
  species: string;
  breed: string;
  birth: Date;
  death?: Date;
  adoption: Date;
  colors: string[];
  weights: Weight[];
  eventAnnotations: EventAnnotation[];
  imgContent: string;
}
